calico_cni/policy_drivers.py

```diff
@@ -256,7 +261,20 @@ def _get_api_pod(self):
             # Perform the API query and handle the result.
             try:
                 _log.debug('Querying Kubernetes API for Pod: %s', path)
-                response = session.get(path, verify=False)
+
```

Comment:
Can we log a clear error message when self.certificate_authority is not None, but the provided API root doesn't use a domain name?  We need to make it clear when there is a configuration problem so that users can fix it on their own.